src/main/java/ca/on/oicr/pineryreports/Main.java

```diff
@@ -0,0 +1,149 @@
+package ca.on.oicr.pineryreports;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+import ca.on.oicr.pinery.client.HttpResponseException;
+import ca.on.oicr.pinery.client.PineryClient;
+import ca.on.oicr.pineryreports.data.ReportFormat;
+import ca.on.oicr.pineryreports.reports.Report;
+import ca.on.oicr.pineryreports.reports.impl.GeccoReport;
+import ca.on.oicr.pineryreports.reports.impl.StockReport;
+
+
+public class Main {
+  
+  public static void main(String[] args) {
+    Options opts = getMainOptions();
+    Report report = null;
+    try {
+      CommandLine mainCommand = getCommandLine(args, opts, false);
+      
+      String pineryOpt = mainCommand.getOptionValue("s");
+      try (PineryClient pinery = new PineryClient(pineryOpt)) {
+        String reportOpt = mainCommand.getOptionValue("r");
+        report = getReport(reportOpt);
+        
+        String formatOpt = mainCommand.getOptionValue("f");
+        ReportFormat format = formatOpt == null ? report.getDefaultFormat() : ReportFormat.get(formatOpt);
+        if (!report.getValidFormats().contains(format)) {
+          throw new ParseException("Invalid format for this report: " + formatOpt);
+        }
+        
+        String outputOpt = mainCommand.getOptionValue("o", "report" + format.getExtension());
+        File outFile = new File(outputOpt);
+        if (outFile.exists()) {
+          throw new ParseException("Output file already exists: " + outputOpt);
+        }
+        
+        for (Option opt : report.getOptions()) {
+          opts.addOption(opt);
+        }
+        CommandLine reportCommand = getCommandLine(args, opts, true);
+        report.processOptions(reportCommand);
+        
+        System.out.println("Options ok. Generating " + report.getTitle() + "...");
+        report.generate(pinery, format, outFile);
+        System.out.println("Report generated: " + outFile.getName());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)